### PR TITLE
recalc with updated SH12A

### LINKED ISSUE #8: scoring target DLET for plan 2 strangely low
and varies wildly to TLET
also primary number is not constant

plan 4a and plan4b also seems to have similar problems.

### PR DESCRIPTION
**Pull request by nbassler**
Recalculation and testing of fix for https://gitlab.au.dk/shieldhit/shieldhit/-/issues/439

### CODE REVIEW

**Comment by nbassler** [comment]
v1.0.1-45-g9c9cc584 introduces new issue https://github.com/APTG/2022_DCPT_LET_timepix/commit/fd04f1b440b9a96ae26dfb0ee4f40808a4e3eca5#diff-6c2d8efdda9ba26991b3ebbf0d0655045dc879e3a3f1b116cd4cf00980607e77
NANs in water scorer.

data/sh12a/results/plan2/NB_Z_narrow_dose_water__p2.png

**Comment by nbassler** [comment]
debugging

```
     Primary particle no.      281000      ETR:   0 hours 29 minutes 13 seconds
*** Warning:  SH_MDEDX_WRAP: z > 20 unavailable. (z,a):          21          39
*** Warning:  SH_MDEDX_WRAP: z > 20 unavailable. (z,a):          21          39
---------------------------------------------------Print particle:
  jpart  = 25
  PDG    = 1000210390
  id     = 1
  Z      = 21
  A      = 39
  amass  =  36314.09731 MeV/c**2
  amu    =     38.98479 u
  gen    = 1
  nprim  = 281484
  weight = 1.000000
POINT Override settings: 4 1.000000 -nan
POINT val = energy * volinv / rho_med * mstpr = 1.230358 * 2.531646 / 1.032000 * -nan =  -nan
POINT NAN in scoring
```
The nans are due to missing stopping power tables for fragments with z > 21. (solid water medium has some heavier elements added)

**Comment by nbassler** [comment]
fixed in SH12A v1.0.1-49-gb2876ad9

**Comment by nbassler** [comment]
Issue #3 primaries still weird, but uncertainties are probably quite ok. @mdsainth maybe take a quick look, you can average the 20 bin target bins if it helps.

If you agree that the test plan looks OK, lets recalculate all plans, but as you suggested for a single target bin, and then merge this branch into master.

filed here: https://github.com/DataMedSci/pymchelper/issues/718

**Comment by nbassler** [comment]
Is queued for recalculation, but will be closed when https://github.com/APTG/2022_DCPT_LET_timepix/issues/6 is fixed too.

**Comment by nbassler** [comment]
FYI: average of the same 200 jobs with a new test version of  pymchelper

https://github.com/DataMedSci/pymchelper/pull/723

now primaries reported should work and be consistent. 

Error bars are however missing now, still TODO.

**Comment by nbassler** [comment]
@mdsainth errorbars there now ... tell us if you see something weird

**Comment by nbassler** [comment]
https://github.com/DataMedSci/pymchelper/pull/723
commit https://github.com/DataMedSci/pymchelper/pull/723/commits/d6c5ac3a5f85b5d74381dac325c83a5a3d584c2d

File: data/sh12a/results/plan2/NB_Z_narrow_LET__p2.dat [hunk 1]
```diff
@@ -1,191 +1,191 @@
...
-15.4756 135.918 0.0669189
-15.5744 144.78 0.0998813
-15.6731 152.886 0.159295
-15.7719 159.776 0.316835
-15.8706 167.534 0.578314
-15.9694 175.267 1.47394
-16.0681 170.647 6.68979
-16.1669 56.3363 7.16404
-16.2656 9.75748 4.44014
+-2.20063 5.2816 3.76115e-06
```
Comment (nbassler) [inline data/sh12a/results/plan2/NB_Z_narrow_LET__p2.dat:1]: Interesting, a slight change in LET for primary protons. 1%

File: data/sh12a/results/plan4_b/NB_target_pos2_L__p05.txt [hunk 2]
```diff
@@ -1,16 +1,7 @@
...
--0.1250000E+02  0.0000000E+00  0.0000000E+00  0.1849294931990758E+02  0.3480631486131116E+01
--0.1230000E+02  0.0000000E+00  0.0000000E+00  0.2015703551083805E+02  0.4257549730084428E+01
--0.1210000E+02  0.0000000E+00  0.0000000E+00  0.2129502671219837E+02  0.4695263679652776E+01
--0.1190000E+02  0.0000000E+00  0.0000000E+00  0.2543096935633871E+02  0.4579901308789938E+01
--0.1170000E+02  0.0000000E+00  0.0000000E+00  0.3027741579393200E+02  0.6228027715036014E+01
--0.1150000E+02  0.0000000E+00  0.0000000E+00  0.3027426487444838E+02  0.5683144248055206E+01
--0.1130000E+02  0.0000000E+00  0.0000000E+00  0.2648653731447320E+02  0.4664163851326128E+01
--0.1110000E+02  0.0000000E+00  0.0000000E+00  0.2299516199194048E+02  0.4018394123904173E+01
+#   PRIMARIES: 0.552D+09
+-0.1200000E+02  0.0000000E+00  0.0000000E+00  0.9273802791247263E+02  0.6636522987769005E+01
```
Comment (nbassler) [inline data/sh12a/results/plan4_b/NB_target_pos2_L__p05.txt:7]: maximum proton dE/dx in PMMA is 9.339e+02 according to PSTAR

**Review by nbassler — COMMENTED** [review]
This looks OK to me

File: data/sh12a/results/plan4_b/NB_target_pos2_L__p05.txt (hunk 2 above)
Comment (nbassler) [reply data/sh12a/results/plan4_b/NB_target_pos2_L__p05.txt:7]: so this is OK

File: data/sh12a/results/plan2/NB_Z_narrow_LET__p2.dat (hunk 1 above)
Comment (nbassler) [reply data/sh12a/results/plan2/NB_Z_narrow_LET__p2.dat:1]: this makes sense. In previous version of SH12A heavy ions (with gen=0 which are target nucleus put into motion elastically) where wrongly scored as protons, which inflated LET a bit.

Comment (grzanka) [reply data/sh12a/results/plan2/NB_Z_narrow_LET__p2.dat:1]: I agree !

**Review by grzanka — APPROVED** [review]

**Pull request merged**